spikeextractors/extractors/neuroscopeextractors/neuroscopeextractors.py

```diff
@@ -0,0 +1,451 @@
+        unit_ids = sorting.get_unit_ids()
+        if len(unit_ids) > 0:
+            spiketrains = [sorting.get_unit_spike_train(u) for u in unit_ids]
+            res = np.concatenate(spiketrains).ravel()
+            clu = np.concatenate([np.repeat(i+1,len(st)) for i,st in enumerate(spiketrains)]).ravel() # i here counts from 0
+            res_sort = np.argsort(res)
+            res = res[res_sort]
+            clu = clu[res_sort]
+        else:
+            res = []
+            clu = []
+        
+        unique_ids = np.unique(clu)
+        n_clu = len(unique_ids)
+            
+        clu = np.insert(clu, 0, n_clu) # The +1 is necessary here b/c the convention for the base sorting object is from 1,...,nUnits
+
+        np.savetxt(save_res, res, fmt='%i')
+        np.savetxt(save_clu, clu, fmt='%i')
+        
+        # create parameters file if none exists
+        if not os.path.isfile(save_xml):
+            soup = BeautifulSoup("",'xml')
+
+            new_tag = soup.new_tag('samplingrate')
+            new_tag.string = str(sorting.get_sampling_frequency())
+            soup.append(new_tag)
+
+            # write parameters file
+            f = open(save_xml, "w")
+            f.write(str(soup))
+            f.close()
+
+
+class NeuroscopeMultiSortingExtractor(MultiSortingExtractor):
+
+    """
+    Extracts spiking information from an arbitrary number of .res.%i and .clu.%i files in the general folder path.
+    
+    The .res is a text file with a sorted list of all spiketimes from all units displayed in sample (integer '%i') units.
+    The .clu file is a file with one more row than the .res with the first row corresponding to the total number of unique ids in the file (and may exclude 0 & 1 from this count)
+    with the rest of the rows indicating which unit id the corresponding entry in the .res file refers to.
+    
+    In the original Neuroscope format:
+        Unit ID 0 is the cluster of unsorted spikes (noise).
+        Unit ID 1 is a cluster of multi-unit spikes.
+        
+    The function defaults to returning multi-unit activity as the first index, and ignoring unsorted noise.
+    To return only the fully sorted units, set keep_mua_units=False.
+        
+    The sorting extractor always returns unit IDs from 1, ..., number of chosen clusters.
+
+    Parameters
+    ----------
+    folder_path : str
+        Optional. Path to the collection of .res and .clu text files. Will auto-detect format.
+    keep_mua_units : bool
+        Optional. Whether or not to return sorted spikes from multi-unit activity. Defaults to True.
+    exclude_inds : list
+        Optional. List of indices to ignore. The set of all possible indices is chosen by default, extracted as the final integer of all the .res.%i and .clu.%i pairs.
+    """
+    extractor_name = 'NeuroscopeMultiSortingExtractor'
+    installed = True  # check at class level if installed or not
+    is_writable = True
+    mode = 'custom'
```

Comment:
this mode should be `folder` (accepted modes are `folder`, `file`, `custom`)

Reply:
What does this specify? If the NeuroscopeRecordingExtractor is adjusted to take a file path instead of a folder path, should I change that to mode = 'file' as well?